admin/class-yoast-notification-center.php

```diff
@@ -522,14 +529,30 @@ public function deactivate_hook() {
 	 */
 	public function update_storage() {
 
-		$notifications = $this->get_notifications();
+		$notifications = $this->notifications;
+
+		/**
+		 * @var Yoast_Notification[] $merged_notifications
+		 */
+		$merged_notifications = array();
+		if ( ! empty( $notifications ) ) {
+			$merged_notifications = call_user_func_array( 'array_merge', $notifications );
+		}
 
 		/**
 		 * Filter: 'yoast_notifications_before_storage' - Allows developer to filter notifications before saving them.
 		 *
 		 * @api Yoast_Notification[] $notifications
 		 */
-		$notifications = apply_filters( 'yoast_notifications_before_storage', $notifications );
+		$merged_notifications = apply_filters( 'yoast_notifications_before_storage', $merged_notifications );
+
+
+		// Split notifications again on user ID.
+		$split_notifications = array();
+		foreach ( $merged_notifications as $notification ) {
+			$split_notifications[ $notification->get_user_id() ][] = $notification;
+		}
+		$notifications = $split_notifications;
```

Comment:
I would move all the logic above to its own method like `format_notifications`. Because this method (`update_storage`) now has multiple responsibilities.